jupyterlab/handlers/yjs_echo_ws.py

```diff
@@ -101,30 +114,94 @@ def on_message(self, message):
         elif message[0] == ServerMessageType.RENAME_SESSION:
             # We move the room to a different entry and also change the room_id property of each connected client
             new_room_id = message[1:].decode("utf-8")
-            for client_id, (loop, hook_send_message, client) in room.clients.items() :
+            for client_id, (loop, hook_send_message, client) in room.clients.items():
                 client.room_id = new_room_id
-            cls.rooms.pop(room_id)
-            cls.rooms[new_room_id] = room
+            ROOMS.pop(room_id)
+            ROOMS[new_room_id] = room
             # print("renamed room to " + new_room_id + ". Old room name was " + room_id)
         elif room:
-            for client_id, (loop, hook_send_message, client) in room.clients.items() :
-                if self.id != client_id :
+            if message[0] == 0:  # sync message
+                read_sync_message(self, room.ydoc.ydoc, message[1:])
+            for client_id, (loop, hook_send_message, client) in room.clients.items():
+                if self.id != client_id:
                     loop.add_callback(hook_send_message, message)
 
     def on_close(self):
         # print("[YJSEchoWS]: close")
-        cls = self.__class__
-        room = cls.rooms.get(self.room_id)
+        room = ROOMS.get(self.room_id)
         room.clients.pop(self.id)
-        if len(room.clients) == 0 :
-            cls.rooms.pop(self.room_id)
+        if not room.clients:
+            ROOMS.pop(self.room_id)
```

Comment:
I would say that this kind of issues already exist (before this PR), right? If you agree, could we tackle them in another PR? I think this PR has been around long enough and it's time to merge it, since it will solve [a critical issue](https://github.com/jupyterlab/jupyterlab/issues/12154).

Reply:
Yeah, that's true. Although we did solve *case 1* with the lock approach. However, there are regular disconnects in the beginning of a session. So I suggest that we at least debounce the destruction of the room after all clients disconnected. This shouldn't be much work. WDYT?

Reply:
Aside from that, I don't see any issues with this PR.

Reply:
Let's do the debouncing in another PR.